scheduler-driver/src/main/java/gov/nasa/jpl/aerie/scheduler/simulation/SimulationFacade.java

```diff
@@ -197,19 +216,16 @@ public static class SimulationException extends Exception {
   }
 
   public void simulateActivity(final SchedulingActivityDirective activity) throws SimulationException {
-    final var activityIdSim = new ActivityDirectiveId(itSimActivityId++);
-    final var activityDirective = schedulingActToActivityDir(activity);
-
-    planActDirectiveIdToSimulationActivityDirectiveId.put(activity.getId(), activityIdSim);
-    driver.simulateActivity(activityDirective, activityIdSim);
-    insertedActivities.put(activity, activityDirective);
+    if(insertedActivities.containsKey(activity)) return;
+    simulateActivities(List.of(activity));
   }
 
   public void computeSimulationResultsUntil(final Duration endTime) {
     var endTimeWithMargin = endTime;
     if(endTime.noLongerThan(Duration.MAX_VALUE.minus(MARGIN))){
       endTimeWithMargin = endTime.plus(MARGIN);
     }
+    //Simulation might throw an exception but we do not catch it because there is no countermeasure
```

Comment:
Could you please elaborate on this? Additionally, if there is an Exception here, why do we not want to catch and rethrow it as a SimulationException?

Reply:
When we call this function, it is not because we have just inserted an activity that we could remove. If there is an exception, we do not know what to do but to crash. We could catch it and rethrow it as a SimulationException though, you're right.